### ISSUE TITLE
[TT-1991] - Unable to install tyk-dashboard, some files are missing

### ISSUE THREAD

**Issue by opensubtitles** [TykTechnologies/tyk#3053]
**Branch/Environment/Version**
- Branch/Version: Master
- Environment: On-prem

**Describe the bug**
`https://packagecloud.io/tyk/tyk-dashboard/packages/ubuntu/bionic/tyk-dashboard_1.9.4.1_amd64.deb?page=1 `
are listed for example files like
`/opt/tyk-dashboard/portal/templates/swagger.html`
When I install it, they are not present on file system

**Reproduction steps**
```
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic

# cat /etc/apt/sources.list.d/tyk_tyk-dashboard.list
deb https://packagecloud.io/tyk/tyk-dashboard/ubuntu/ bionic main
deb-src https://packagecloud.io/tyk/tyk-dashboard/ubuntu/ bionic main

# ls /opt/tyk-dashboard
ls: cannot access '/opt/tyk-dashboard': No such file or directory

# apt install tyk-dashboard
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  tyk-dashboard
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/30.2 MB of archives.
After this operation, 68.5 MB of additional disk space will be used.
Selecting previously unselected package tyk-dashboard.
(Reading database ... 222966 files and directories currently installed.)
Preparing to unpack .../tyk-dashboard_1.9.4.1_amd64.deb ...
Creating user and group...
Unpacking tyk-dashboard (1.9.4.1) ...
Setting up tyk-dashboard (1.9.4.1) ...
Setting permissions
chmod: cannot access '/opt/tyk-dashboard/tyk_analytics.conf': No such file or directory
Installing init scripts...
Found Systemd

# ls /opt/tyk-dashboard/portal/templates/
dev
```

As you can see, the files for templates are not there by default installation according docs. I thought error could be error message
`chmod: cannot access '/opt/tyk-dashboard/tyk_analytics.conf': No such file or directory`
so I run 
`sudo /opt/tyk-dashboard/install/setup.sh [parameters]`

then remove tyk-dashboard and install it again, now the error 
`chmod: cannot access '/opt/tyk-dashboard/tyk_analytics.conf': No such file or directory`
is gone, but anyway, there are no files in 
```
# ls /opt/tyk-dashboard/portal/templates/
dev
```
(dev directory is empty). 

I am not sure what I am doing wrong and I would really like to try TYK, please help

(similar problem is with tyk-gateway - files are missing there)

**Comment by alephnull** [comment]
This is very strange. I am unable to reproduce on a fresh bionic install. Test log below.
However, this test is configured to setup packagecloud like `curl -s https://packagecloud.io/install/repositories/tyk/${repo}/script.deb.sh | bash` rather than manually setting up files in `sources.list.d`. You could do the same and move on while I figure out if the static instructions are incorrect.

```
After this operation, 68.5 MB of additional disk space will be used.
Get:1 https://packagecloud.io/tyk/tyk-dashboard/ubuntu bionic/main amd64 tyk-dashboard amd64 1.9.4.1 [30.2 MB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 30.2 MB in 7s (4251 kB/s)
Selecting previously unselected package tyk-dashboard.
(Reading database ... 4787 files and directories currently installed.)
Preparing to unpack .../tyk-dashboard_1.9.4.1_amd64.deb ...
Creating user and group...
Unpacking tyk-dashboard (1.9.4.1) ...
Setting up tyk-dashboard (1.9.4.1) ...
Setting permissions
Installing init scripts...
Found SysV1
Removing intermediate container fe72de608a01
 ---> 9e0305c4fb5f
Successfully built 9e0305c4fb5f
Successfully tagged signing_dashboard-debian:latest
WARNING: Image for service dashboard-debian was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
/opt/tyk-dashboard/portal/templates:
blueprint.html	    member_navigation.html  request_key_success.html
blueprint_docs.html  navigation.html	    request_multi_key.html
catalogue.html	    oauth_client.html	    request_multi_key_success.html
dev		    oauth_home.html	    scripts.html
footer.html	    page.html		    secondary_navigation.html
header.html	    password_reset.html     signup.html
index.html	    profile.html	    signup_success.html
login_page.html      raw.html		    swagger.html
member_home.html     request_key.html

/opt/tyk-dashboard/portal/templates/dev:
aglio_test.html
```

**Comment by alephnull** [comment]
@marksou Do you think it would be appropriate to ask customers to configure package cloud for use with the script? The script would keep up with changes.

**Comment by excieve** [comment]
@alephnull Sorry, not sure how is script different? It all ends up in the standard repo definition at `/etc/apt/sources.list.d`, doesn't it?

**Comment by alephnull** [comment]
> @alephnull Sorry, not sure how is script different? It all ends up in the standard repo definition at `/etc/apt/sources.list.d`, doesn't it?

There are two differences that I can see:
1. For Debian/Ubuntu, it will detect the release and add it to the sources.list entry.
2. For RHEL <8, it does not enable gpgcheck

But the real advantage is that it sets up all the dependencies required to use packagecloud on that host. It can also handle RHEL8 (it enables gpgcheck there).

My proposal is to use the script to configure the repository and then add a section on enabling gpgcheck just for RHEL/CentOS <8.

**Comment by opensubtitles** [comment]
I tried again, but as I thought, it will have same result:
```
# ls -la /etc/apt/sources.list.d
total 0
drwxr-xr-x 2 root root   6 May  4 06:12 .
drwxr-xr-x 7 root root 222 Apr 29 09:04 ..

# curl -s https://packagecloud.io/install/repositories/tyk/tyk-dashboard/script.deb.sh | sudo bash
Detected operating system as Ubuntu/bionic.
Checking for curl...
Detected curl...
Checking for gpg...
Detected gpg...
Running apt-get update... done.
Installing apt-transport-https... done.
Installing /etc/apt/sources.list.d/tyk_tyk-dashboard.list...done.
Importing packagecloud gpg key... done.
Running apt-get update... done.

The repository is setup! You can now install packages.

# less tyk_tyk-dashboard.list
# this file was generated by packagecloud.io for
# the repository at https://packagecloud.io/tyk/tyk-dashboard

deb https://packagecloud.io/tyk/tyk-dashboard/ubuntu/ bionic main
deb-src https://packagecloud.io/tyk/tyk-dashboard/ubuntu/ bionic main

# apt install tyk-dashboard
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  tyk-dashboard
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/30.2 MB of archives.
After this operation, 68.5 MB of additional disk space will be used.
Selecting previously unselected package tyk-dashboard.
(Reading database ... 222953 files and directories currently installed.)
Preparing to unpack .../tyk-dashboard_1.9.4.1_amd64.deb ...
Creating user and group...
Unpacking tyk-dashboard (1.9.4.1) ...
Setting up tyk-dashboard (1.9.4.1) ...
Setting permissions
chmod: cannot access '/opt/tyk-dashboard/tyk_analytics.conf': No such file or directory
Installing init scripts...
Found Systemd

# ls -la /opt/tyk-dashboard/portal/templates/
total 0
drwxr-xr-x 3 tyk tyk 17 May  4 06:17 .
drwxr-xr-x 6 tyk tyk 79 May  4 06:17 ..
drwxr-xr-x 2 tyk tyk  6 May  4 06:17 dev

#########################
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic

# find /opt/tyk-dashboard
/opt/tyk-dashboard
/opt/tyk-dashboard/utils
/opt/tyk-dashboard/utils/scripts
/opt/tyk-dashboard/utils/scripts/import_policies.sh
/opt/tyk-dashboard/utils/scripts/import_org.sh
/opt/tyk-dashboard/utils/scripts/import_apis.sh
/opt/tyk-dashboard/utils/scripts/export.sh
/opt/tyk-dashboard/portal
/opt/tyk-dashboard/portal/templates
/opt/tyk-dashboard/portal/templates/dev
/opt/tyk-dashboard/portal/portal-assets
/opt/tyk-dashboard/portal/portal-assets/fonts
/opt/tyk-dashboard/portal/portal-assets/images
/opt/tyk-dashboard/portal/portal-assets/css
/opt/tyk-dashboard/portal/portal-assets/js
/opt/tyk-dashboard/portal/email_templates
/opt/tyk-dashboard/portal/jade
/opt/tyk-dashboard/schemas
/opt/tyk-dashboard/install
/opt/tyk-dashboard/install/data
/opt/tyk-dashboard/install/data/portal.json
/opt/tyk-dashboard/install/data/tyk_analytics_template.conf
/opt/tyk-dashboard/install/data/portal_api.json
/opt/tyk-dashboard/install/data/portal_assets.json
/opt/tyk-dashboard/install/inits
/opt/tyk-dashboard/install/inits/upstart
/opt/tyk-dashboard/install/inits/upstart/init
/opt/tyk-dashboard/install/inits/upstart/init/1.x
/opt/tyk-dashboard/install/inits/upstart/init/1.x/tyk-dashboard.conf
/opt/tyk-dashboard/install/inits/upstart/init/0.x
/opt/tyk-dashboard/install/inits/upstart/init/0.x/tyk-dashboard.conf
/opt/tyk-dashboard/install/inits/upstart/default
/opt/tyk-dashboard/install/inits/upstart/default/tyk-dashboard
/opt/tyk-dashboard/install/inits/sysv
/opt/tyk-dashboard/install/inits/sysv/init.d
/opt/tyk-dashboard/install/inits/sysv/init.d/tyk-dashboard
/opt/tyk-dashboard/install/inits/sysv/default
/opt/tyk-dashboard/install/inits/sysv/default/tyk-dashboard
/opt/tyk-dashboard/install/inits/systemd
/opt/tyk-dashboard/install/inits/systemd/system
/opt/tyk-dashboard/install/inits/systemd/system/tyk-dashboard.service
/opt/tyk-dashboard/install/inits/systemd/default
/opt/tyk-dashboard/install/inits/systemd/default/tyk-dashboard
/opt/tyk-dashboard/install/post_install.sh
/opt/tyk-dashboard/install/post_remove.sh
/opt/tyk-dashboard/install/setup.sh
/opt/tyk-dashboard/install/bootstrap.sh
/opt/tyk-dashboard/install/init_local.sh
/opt/tyk-dashboard/install/before_install.sh
/opt/tyk-dashboard/lang
/opt/tyk-dashboard/tyk-analytics
/opt/tyk-dashboard/CHANGELOG.md
/opt/tyk-dashboard/EULA.md
/opt/tyk-dashboard/README.md

# find /opt/tyk-dashboard | wc -l
54
```

If I can turn on some debug mode to fix this, let me know. It is very strange. Maybe I can restart machine (will do now)

**Comment by alephnull** [comment]
The installation problem has been resolved since 4.0 at least. 1.9.1 is not supported any more.

Please open a fresh issue for any remaining problems.
